### ISSUE TITLE
URL Parser does not look for SRV DNS records

### ISSUE THREAD

**Issue by rafaelm7o** [ankhers/mongodb#216]
I'm trying to connect to MongoDB Atlas but I always get connection refused

This is my URL

```
config :order_db, :db,
  url: "mongodb+srv://USER:PASSWORD@cluster0-XXXX.mongodb.net/order_dev",
```

And the call is the following:
` {:ok, conn} = Mongo.start_link(url: Application.get_env(:order_db, :db)[:url])`

Any idea what's going on?

**Comment by ankhers** [comment]
Can you please share which version of the driver you are using? Can you also show the error that you are receiving?

**Comment by rafaelm7o** [comment]
Hi ankhers

I tried with

```
      {:mongodb, github: "ankhers/mongodb", branch: "master"},
```
and

```
      {:mongodb, ">= 0.0.0"},
```

By the way, MongoDB Atlas uses the 3.6 version

The specific error is:

> [error] Mongo.Protocol (#PID<0.640.0>) failed to connect: ** (Mongo.Error) tcp connect: connection refused - :econnrefused

PD. I created a mongodb server with 3.4 and it worked correctly

**Comment by ankhers** [comment]
It looks like you have `+srv` in your url which is meant to tell the driver to look at SRV DNS records[1]. I believe that is not currently implemented in this drivers URL Parser. 

If you, or anyone else, would care to take a crack at adding this, I would be more than happy to help.

[1] https://docs.mongodb.com/manual/reference/connection-string/#dns-seedlist-connection-format

**Comment by rafaelm7o** [comment]
I've created the implementation in the url_parser for the mongodb+srv, let'me create a PR for the changes

**Comment by ankhers** [comment]
Merged #219
